client/src/main/java/io/pravega/client/stream/impl/ReaderGroupStateManager.java

```diff
@@ -114,7 +113,7 @@ void initializeReader() {
             throw new IllegalStateException("The requested reader: " + readerId
                     + " cannot be added to the group because it is already in the group. Perhaps close() was not called?");
         }
-        acquireTimer.zero();
+        acquireTimer.reset(Duration.ofMillis(initialAllocationDelay));
```

Comment:
Please consider making this delay random (from 0 to initialAllocationDelay) otherwise it will be exactly same as all the readers starting after initialAllocationDelay.